File: Backend/src/main/java/com/example/project/dto/request/OrderedProductRequest.java
package com.example.project.dto.request;


import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class OrderedProductRequest {
    @JsonProperty("product_id")
    @NotNull(message = "ID продукта не может быть пустым")
    private Integer productId;

    @JsonProperty("count")
    @NotNull(message = "Количество продукта не может быть пустым")
    @PositiveOrZero(message="Кол-во продукта должно быть положительным числом")
    private Integer countProduct;
}

File: Android/app/src/main/java/com/kire/market_place_android/data/remote/dto/request/product/OrderedProductRequest.kt
package com.kire.market_place_android.data.remote.dto.request.product

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class OrderedProductRequest(
    @SerialName("product_id")
    val productId: Int,
    @SerialName("count")
    val countProduct: Int
)